Expensify/App
# Fix broken duplicate name error in User Created Policy Rooms

@pecanoro, please review when you get the chance.

### Details
Fixes the bug where we don't show an error if you try to create a policy room with the same name as one just created. What was happening was that we weren't checking for duplicates after a policy was selected - we were only doing it when the room name's text changed. This will fix that and show the error in the WorkspaceNew component.


### Fixed Issues

$ https://github.com/Expensify/App/issues/7260

### Tests/WebQA

1. Go to https://staging.new.expensify.com/
2. Create room e.g. "fish" (Use Green + on the bottom left. You need to have a workspace also)
3. Create second room with same name "fish". First type the name fish in and THEN select a workspace.
![image](https://user-images.githubusercontent.com/19364431/151890724-8f8a7b6c-3b90-4e45-8b7d-348a536e33cc.png)
![image](https://user-images.githubusercontent.com/19364431/151890735-b3c4bf50-d7e4-4764-97d5-8cc457dd246b.png)

4. An error should show up after you select the workspace and it should prevent you from clicking "Create Room"
![image](https://user-images.githubusercontent.com/19364431/151890755-7d87e329-6b0b-4c96-b5c7-642948db21d6.png)
![image](https://user-images.githubusercontent.com/19364431/151890764-980730c6-3af7-4fd2-bbbe-4f51d109dd22.png)




### Tested On
